src/util/string.cpp

```diff
@@ -743,3 +743,161 @@ void str_replace(std::string &str, char from, char to)
 {
 	std::replace(str.begin(), str.end(), from, to);
 }
+
+std::wstring translate_all(const std::wstring &s, size_t &i);
+
+std::wstring translate_string(const std::wstring &s, size_t &i) {
```

Comment:
It seems to me that since i is a `size_t`, copying is better than using a pointer, am I wrong?